community/values/src/main/java/org/neo4j/values/storable/TimeConstants.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.values.storable;
+
+import static java.time.temporal.ChronoUnit.DAYS;
+
+@SuppressWarnings( "WeakerAccess" )
+public class TimeConstants
+{
+    public static final long NANOS_PER_SECOND = 1_000_000_000L;
```

Comment:
Why are the ones that do fit in an int still defined as a long, requiring casting elsewhere?

Reply:
Leaving `NANOS_PER_SECOND` as an `int` gives some interesting overflow problems if you multiply it with another `int`, e.g. `offsetSeconds`. I find these problems much harder to detect, than it is annoying to down-cast to `int` in the 1 place where that is required. Thus this.